### DISCUSSION TITLE
FATAL: Module binder_linux not found in directory

### DISCUSSION THREAD

**Discussion by sunwoo1524** [waydroid/waydroid#1175] [Q&A]
I was installing waydroid following [this arch wiki document](https://wiki.archlinux.org/title/Waydroid).

But, when I ran `sudo waydroid init`, there was an error:
> [13:43:23] Failed to load binder driver
> [13:43:23] modprobe: FATAL: Module binder_linux not found in directory /lib/modules/6.6.2-arch1-1
> [13:43:23] ERROR: Binder node "binder" for waydroid not found
> [13:43:23] See also: <https://github.com/waydroid>
> Run 'waydroid log' for details.

I think I already installed binder_linux.

Please tell me what I should I do...

**Comment by pwl1m** [comment]
i have the same issue, do you resolve? @sunwoo1524

**Reply by sunwoo1524** [reply]
> i have the same issue, do you resolve? @sunwoo1524

Oh sorry, the reply is so late. It had not resolved. And, unfortunately, I removed my arch linux and I'm using windows now because of some private reason.

**Comment by cniw** [comment]
> Failed to load binder driver
> modprobe: FATAL: Module binder_linux not found in directory /lib/modules/6.6.2-arch1-1
> ERROR: Binder node "binder" for waydroid not found

Default Arch Linux kernel doesn't has the binder module, to solve this issue you can either side-load the module to your kernel or install another kernel that has binder module eg. linux-zen, linux-xanmod, etc.

  1. `yay -S binder_linux-dkms` you can replace `yay` with `paru` or the other AUR helper program you have, upstream link https://github.com/choff/anbox-modules. To make sure it installed correctly check with `dkms status`
  2. `sudo pacman -S linux-zen linux-zen-headers` or `yay -S linux-xanmod linux-xanmod-headers` then you should reboot your PC and start using that new kernel, select it on your boot manager list menu. To make sure you're on the new kernel check with `uname -sr`

**Comment by jkpjkpjkp** [comment]
to change the subject completely,

does anyone know how to fix this on Pop!OS 24.04 (COSMIC)?

**Reply by ZanDev32** [reply]
Most of the distro like Pop!OS and ZorinOS doesn't ship their stock kernel with binder module enabled, There are 3 ways to solve this issue:

1. Using the [DKMS](https://github.com/choff/anbox-modules) for binder_linux and load it to the stock kernel
2. Using pre-built custom kernel with binder_linux module enabled (e.g  [linux-xanmod](https://xanmod.org/) or [linux-liquorix](https://liquorix.net/))
3. Building your own kernel (not recommended)

**Comment by tomchiverton** [comment]
Broken on Ubuntu too. This all needs adding to the docs at the very least.

I don't see a binder DKMS package in the stock repos

**Comment by tomchiverton** [comment]
Waydroid is deader than sideways flares with pants in the knees
-- 
†øღ
